Tests/web3swiftTests/localTests/web3swiftTransactionsTests.swift

```diff
@@ -1,48 +1,627 @@
-                                                  r: BigUInt(0),
-                                                  s: BigUInt(0))
+            var jsonTxn: EthereumTransaction = try JSONDecoder().decode(EthereumTransaction.self, from: jsonData)
+
+            // unsign the input transaction
+            jsonTxn.unsign()
+
+            // now sign the transaction with the private key
+            try jsonTxn.sign(privateKey: privateKeyData, useExtraEntropy: false)
+            let rawTxn = jsonTxn.encode()!.toHexString().addHexPrefix()
+
+            // check the hash, if they match everything was parsed, and re-encoded correctly
+            XCTAssert(rawTxn == vector.RLP)
+            // check the hash, if they match everything was parsed, and re-encoded correctly
+            XCTAssert(jsonTxn.hash!.toHexString().addHexPrefix() == vector.hash)
+            // check that we recovered the address correctly
+            XCTAssert(jsonTxn.sender!.address == expectedAddress.address)
+        } catch {
+            print(error)
+            return XCTFail(String(describing: error))
+        }
+    }
+
+    // test EIP-1559 encoding with an access list
+    func testEIP1559WithAccessDecodeAndRecoverJSON() throws {
+        let vector = testVector[TestCase.eip1559withAccessList.rawValue]
+
+        let expectedAddress = EthereumAddress(testSenderAddress)!
+        let jsonData = vector.JSON.data(using: .utf8)!
+
+        do {
+            let jsonTxn: EthereumTransaction = try JSONDecoder().decode(EthereumTransaction.self, from: jsonData)
+            // check that the transaction type is eip1559
+            XCTAssert(jsonTxn.type == .eip1559)
+            // check the hash, if they match everything was parsed, and re-encoded correctly
+            XCTAssert(jsonTxn.hash!.toHexString().addHexPrefix() == vector.hash)
+            // check that we recovered the address correctly
+            XCTAssert(jsonTxn.sender!.address == expectedAddress.address)
+        } catch {
+            print(error)
+            return XCTFail(String(describing: error))
+        }
+    }
+
+    // test EIP-1559 encoding with an access list
+    func testEIP1559WithAccessDecodeAndRecoverRLP() throws {
+        let vector = testVector[TestCase.eip1559withAccessList.rawValue]
+
+        let expectedAddress = EthereumAddress(testSenderAddress)!
+        let rlpData = Data.fromHex(vector.RLP)!
+
+        guard let rlpTxn = EthereumTransaction(rawValue: rlpData) else {
+            return XCTFail("Decoding RLP encoded transaction failed")
+        }
+        // check that the transaction type is eip1559
+        XCTAssert(rlpTxn.type == .eip1559)
+        // check the hash, if they match everything was parsed, and re-encoded correctly
+        XCTAssert(rlpTxn.hash!.toHexString().addHexPrefix() == vector.hash)
+        // check that we recovered the address correctly
+        XCTAssert(rlpTxn.sender!.address == expectedAddress.address)
```

Comment:
It's not an objection but `XCTAssertEqual` is primarily a convenience for comparing the results of expressions.
It's declared as 
```
#define XCTAssertEqual(expression1, expression, ...)
```

Reply:
Hmm, I see that my phrasing or maybe even understanding was not exactly right.
The difference is that `XCTAssert` expects boolean and `XCTAssertEqual` expects expressions of scalar C type.

I'd say both options are good enough 🙂

Reply:
I don't object to the change either. But I think I will leave it as-is for the time being. Leaving the change for a larger refactor of the tests, as this is the pattern currently used everywhere.

Reply:
It's ok to not change it yet, i agree this would be a drop in a sea in term of tests refactoring. But i still suggest to make tests assertion more explicit. Could you please just set todo comment with that suggestion somewhere within a test file?

Reply:
I'll add a `todo:`, and will use the more explicit `XCTAssertEqual` in the new tests I have added. I'll add todo's to the other 3 test files touched in this PR as well.

Reply:
Done, Added the TODO's to the two other files, and changed the asserts in this file in the tests I added. Also added failure messages to the asserts, which I think we should be making sure are always included going forward.